### PR TITLE
Implement tosa.conv2d

### PR DESCRIPTION
**Pull request by inwchun**
Implement tosa.conv2d

### CODE REVIEW

**Comment by inwchun** [comment]
all litmus tests succeed except conv2d2.mlir. This takes 360 seconds, but the result is correct (case of stride 2).

File: src/encode.cpp [hunk 1]
```diff
@@ -970,6 +970,55 @@ void encodeOp(State &st, mlir::tosa::BitwiseXorOp op, bool) {
...
+                            srcDims[2] + pad[2] + pad[3], srcDims[3]};
+
+  auto cond = padInd[1].uge(pad[0]) & padInd[1].ult(pad[0] + srcDims[1]) &
+                padInd[2].uge(pad[2]) & padInd[2].ult(pad[2] + srcDims[2]);
+
+  Expr output = Expr::mkIte(cond, input.get(srcInd).first, *getZero(elemTy))
+                + bias.get({padInd[3]}).first;
+  Expr init = Expr::mkIte(cond, Expr::mkBool(true), input.isInitialized(padInd));
+
+  auto padInput = Tensor::mkLambda(
```
Comment (aqjune) [inline src/encode.cpp:1012]: You can use `Tensor::mkInitializedLambda` here

Comment (aqjune) [reply src/encode.cpp:1012]: Also, please add `st.wellDefined(op, {input, weight, bias}.isFullyInitialized())`

File: tests/long-opts/conv2d-nchw/convolution_nchw.src.mlir [hunk 2]
```diff
@@ -1,10 +0,0 @@
-// VERIFY
```
Comment (aqjune) [inline tests/long-opts/conv2d-nchw/convolution_nchw.src.mlir:1]: Is there a reason why this test has been removed?

Comment (inwchun) [reply tests/long-opts/conv2d-nchw/convolution_nchw.src.mlir:1]: oops. forgot to add file after renaming dir

**Pull request merged**